core/services/ocr2/plugins/functions/aggregation.go

```diff
@@ -35,28 +36,46 @@ func Aggregate(aggMethod config.AggregationMethod, observations []*encoding.Proc
 			successful = append(successful, obs)
 		}
 	}
+	resultIsError := len(errored) > len(successful)
+	var toAggregate []*encoding.ProcessedRequest
 	var rawData [][]byte
-	if len(errored) > len(successful) {
+	if resultIsError {
+		toAggregate = errored
 		for _, item := range errored {
 			rawData = append(rawData, item.Error)
 		}
+	} else {
+		toAggregate = successful
+		for _, item := range successful {
+			rawData = append(rawData, item.Result)
+		}
+	}
+	// All metadata is aggregated using MODE method
```

Comment:
I wonder what the implications of this are. With mode it seems like it will always be choosing at random because it's likely execution metrics will vary.

Can one node have a failure somewhere and report 0's in their metadata, then because of mode that one is chosen to represent the whole DON, so they don't get compensated fairly?

Reply:
MODE is correct for metadata that needs to be constant across all honest nodes like CallbackGas and Commitment. You're correct that stuff like execution metrics shouldn't be aggregated using MODE. I'll change the comment here.